File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.cs
using System;
using System.Collections.Generic;
using System.Text;

public class Solution {
    public string solution(int[] food) {
        StringBuilder answer = new StringBuilder();
        LinkedList<int> deque = new LinkedList<int>();
        
        deque.AddFirst(0);
        int num = food.Length - 1;

        for(int i = food.Length - 1; i >= 1; i--)
        {
            if(food[i] == 1)
            {
                num--;
                continue;
            }
            int count = food[i] / 2;
            for(int j = 0; j < count; j++)
            {
                deque.AddFirst(num);
                deque.AddLast(num);
            }
            num--;
        }
        while(deque.Count > 0 )
        {
            int temp = deque.First.Value;
            deque.RemoveFirst();
            answer.Append(temp);
        }
        return answer.ToString();
    }
}

File: 프로그래머스/1/134240. 푸드 파이트 대회/푸드 파이트 대회.cpp
#include <string>
#include <vector>
#include <bits/stdc++.h>

using namespace std;

string solution(vector<int> food) {
    string answer = "";
    deque<int> dq;
    dq.push_back(0);
    
    for(int i = food.size() - 1; i >= 1; i--)
    {
        if(food[i] / 2 != 0)
        {
            int cnt = food[i] / 2;
            for(int j = 0; j < cnt; j++)
            {
                dq.push_back(i);
                dq.push_front(i);
            }
        }
    }
    
    for(int n : dq)
    {
        answer += to_string(n);
    }
    return answer;
}

// 일단 0을 먼저 넣고, 칼로리 높은 음식부터 좌우로 쌓음, 단, 좌우가 개수가 같아야함